lib/pos_fee_estimator.go

```diff
@@ -126,7 +126,7 @@ func addBlockToTransactionRegister(txnRegister *TransactionRegister, block *MsgD
 		if txn.TxnMeta.GetTxnType() == TxnTypeBlockReward {
 			continue
 		}
-		mtxn, err := NewMempoolTx(txn, block.Header.TstampNanoSecs, block.Header.Height)
+		mtxn, err := NewMempoolTx(txn, NanoSecondsToUint64MicroSeconds(block.Header.TstampNanoSecs), block.Header.Height)
```

Comment:
Should we use Int64 for the microsecond tracking of mempool txs? this could be addressed separately.

Reply:
I'm open to leaving this as-is, switching to int64 microseconds, or to pass in the `Time` struct type. Happy to defer to @AeonSw4n on how he wants to move forward for the Mempool

Reply:
I think let's just keep it as is for simplicity. It shouldn't change how transactions are ordered within the TransactionRegister instance.